### ISSUE TITLE
IngressRouteTCP always routes to same pod (Load balancing doesn't work)

### ISSUE THREAD

**Issue by guyromb** [containous/traefik#6566]
### Do you want to request a *feature* or report a *bug*? bug

Bug

### What did you do?

Created a simple IngressRouteTCP
```
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRouteTCP
metadata:
  name: ingressroutetcp-livestream
spec:
  entryPoints:
    - rtmp
  routes:
  - match: HostSNI(`*`)
    services:
    - name: livestream-service
      port: 1935
      TerminationDelay: 400
```

livestream-service is a regular ClusterIP - points to two Pods (scaled deployment). I do see two Pods in Traefik dashboard so it seems traefik detect that but it always routes to the same Pod. When trying to setup a regular loadbalancer with external IP (and not use traefik) everything works.

### What did you expect to see?

Routing to random pod / loadbalance, as usual

### What did you see instead?

Only routes to one pod regardless of how many pods exists in the service.

### What is your environment & configuration (arguments, toml, provider, platform, ...)? entrypoint is set, and IngressRouteTCP 

```
# traefik.toml
[log]
  level = "DEBUG"
[serversTransport]
  insecureSkipVerify = true
[ping]
[api]
  dashboard = true
  insecure = true
[entryPoints]
  [entryPoints.traefik]
    address = ":9000"
  [entryPoints.web]
    address = ":8000"
  [entryPoints.websecure]
    address = ":8443"
  [entryPoints.rtmp]
    address = ":1935"
[providers.kubernetesCRD]
[providers.kubernetesIngress]
[[tls.certificates]]
  certFile = "/ssl-dev/tls.crt"
  keyFile = "/ssl-dev/tls.key"
```

**Comment by SantoDE** [comment]
Hello @guyromb,

thanks for your interest in the project.

To better triage the bug possible, could you please provide is with your API rawdata? That is accessible on `api/rawdata` instead of `/dashboard`.

Thanks!

**Comment by guyromb** [comment]
```
{
"routers": {
"livestream-service-api-develop-dev-fit-ci-tls@kubernetes": {
"service": "develop-livestream-service-service-3000",
"rule": "Host(`livestream-service-api-develop.dev.fit.ci`) && PathPrefix(`/`)",
"tls": {},
"status": "enabled",
"using": [
"rtmp",
"traefik",
"web",
"websecure"
]
},
"livestream-service-api-develop-dev-fit-ci@kubernetes": {
"service": "develop-livestream-service-service-3000",
"rule": "Host(`livestream-service-api-develop.dev.fit.ci`) && PathPrefix(`/`)",
"status": "enabled",
"using": [
"rtmp",
"traefik",
"web",
"websecure"
]
},
"livestream-service-ms-develop-dev-fit-ci-tls@kubernetes": {
"service": "develop-livestream-service-service-8000",
"rule": "Host(`livestream-service-ms-develop.dev.fit.ci`) && PathPrefix(`/`)",
"tls": {},
"status": "enabled",
"using": [
"rtmp",
"traefik",
"web",
"websecure"
]
},
"livestream-service-ms-develop-dev-fit-ci@kubernetes": {
"service": "develop-livestream-service-service-8000",
"rule": "Host(`livestream-service-ms-develop.dev.fit.ci`) && PathPrefix(`/`)",
"status": "enabled",
"using": [
"rtmp",
"traefik",
"web",
"websecure"
]
},
"ping@internal": {
"entryPoints": [
"traefik"
],
"service": "ping@internal",
"rule": "PathPrefix(`/ping`)",
"priority": 2147483647,
"status": "enabled",
"using": [
"traefik"
]
},
"traefik-traefik-dashboard-960a963b07256eae042e@kubernetescrd": {
"entryPoints": [
"https"
],
"middlewares": [
"traefik-traefik-dashboard-auth"
],
"service": "traefik-traefik-dashboard-960a963b07256eae042e",
"rule": "Host(`traefik2.example.com`)",
"tls": {},
"error": [
"entryPoint \"https\" doesn't exist",
"no valid entryPoint for this router"
],
"status": "disabled"
},
"traefik-v2-traefik-dashboard-d012b7f875133eeab4e5@kubernetescrd": {
"entryPoints": [
"traefik"
],
"service": "api@internal",
"rule": "PathPrefix(`/dashboard`) || PathPrefix(`/api`)",
"status": "enabled",
"using": [
"traefik"
]
},
"traefik-v2-traefik-v2-dashboard-d012b7f875133eeab4e5@kubernetescrd": {
"entryPoints": [
"traefik"
],
"service": "api@internal",
"rule": "PathPrefix(`/dashboard`) || PathPrefix(`/api`)",
"status": "enabled",
"using": [
"traefik"
]
}
},
"middlewares": {
"dashboard_redirect@internal": {
"redirectRegex": {
"regex": "^(http:\\/\\/[^:\\/]+(:\\d+)?)\\/$",
"replacement": "${1}/dashboard/",
"permanent": true
},
"status": "enabled",
"usedBy": [
"dashboard@internal"
]
},
"dashboard_stripprefix@internal": {
"stripPrefix": {
"prefixes": [
"/dashboard/",
"/dashboard"
]
},
"status": "enabled",
"usedBy": [
"dashboard@internal"
]
}
},
"services": {
"api@internal": {
"status": "enabled",
"usedBy": [
"api@internal",
"traefik-v2-traefik-dashboard-d012b7f875133eeab4e5@kubernetescrd",
"traefik-v2-traefik-v2-dashboard-d012b7f875133eeab4e5@kubernetescrd"
]
},
"cypress-website-nginx-service-443@kubernetes": {
"loadBalancer": {
"servers": [
{
"url": "https://10.32.5.68:443"
}
],
"passHostHeader": true
},
"status": "enabled",
"usedBy": [
"web-cypress-app-fit-ci@kubernetes"
],
"serverStatus": {
"https://10.32.5.68:443": "UP"
}
},
"dashboard@internal": {
"status": "enabled",
"usedBy": [
"dashboard@internal"
]
},
"develop-livestream-service-service-3000@kubernetes": {
"loadBalancer": {
"servers": [
{
"url": "http://10.32.13.178:3000"
},
{
"url": "http://10.32.6.94:3000"
}
],
"passHostHeader": true
},
"status": "enabled",
"usedBy": [
"livestream-service-api-develop-dev-fit-ci-tls@kubernetes",
"livestream-service-api-develop-dev-fit-ci@kubernetes"
],
"serverStatus": {
"http://10.32.13.178:3000": "UP",
"http://10.32.6.94:3000": "UP"
}
},
"develop-livestream-service-service-8000@kubernetes": {
"loadBalancer": {
"servers": [
{
"url": "http://10.32.13.178:8000"
},
{
"url": "http://10.32.6.94:8000"
}
],
"passHostHeader": true
},
"status": "enabled",
"usedBy": [
"livestream-service-ms-develop-dev-fit-ci-tls@kubernetes",
"livestream-service-ms-develop-dev-fit-ci@kubernetes"
],
"serverStatus": {
"http://10.32.13.178:8000": "UP",
"http://10.32.6.94:8000": "UP"
}
},
"traefik-traefik-dashboard-960a963b07256eae042e@kubernetescrd": {
"loadBalancer": {
"servers": [
{
"url": "http://10.32.1.10:8080"
}
],
"passHostHeader": true
},
"status": "enabled",
"usedBy": [
"traefik-traefik-dashboard-960a963b07256eae042e@kubernetescrd"
]
}
},
"tcpRouters": {
"default-ingressroutetcp-livestream-673acf455cb2dab0b43a@kubernetescrd": {
"entryPoints": [
"rtmp"
],
"service": "default-ingressroutetcp-livestream-673acf455cb2dab0b43a",
"rule": "HostSNI(`*`)",
"error": [
"the service \"default-ingressroutetcp-livestream-673acf455cb2dab0b43a@kubernetescrd\" does not exist"
],
"status": "disabled",
"using": [
"rtmp"
]
},
"develop-livestream-service-ingressroutetcp-livestream-673acf455cb2dab0b43a@kubernetescrd": {
"entryPoints": [
"rtmp"
],
"service": "develop-livestream-service-ingressroutetcp-livestream-673acf455cb2dab0b43a",
"rule": "HostSNI(`*`)",
"status": "enabled",
"using": [
"rtmp"
]
},
"feature-livestream-prep-livestream-service-ingressroutetcp-livestream-673acf455cb2dab0b43a@kubernetescrd": {
"entryPoints": [
"rtmp"
],
"service": "feature-livestream-prep-livestream-service-ingressroutetcp-livestream-673acf455cb2dab0b43a",
"rule": "HostSNI(`*`)",
"error": [
"the service \"feature-livestream-prep-livestream-service-ingressroutetcp-livestream-673acf455cb2dab0b43a@kubernetescrd\" does not exist"
],
"status": "disabled",
"using": [
"rtmp"
]
}
},
"tcpServices": {
"develop-livestream-service-ingressroutetcp-livestream-673acf455cb2dab0b43a@kubernetescrd": {
"loadBalancer": {
"terminationDelay": 100,
"servers": [
{
"address": "10.32.13.178:1935"
},
{
"address": "10.32.6.94:1935"
}
]
},
"status": "enabled",
"usedBy": [
"develop-livestream-service-ingressroutetcp-livestream-673acf455cb2dab0b43a@kubernetescrd"
]
}
}
}
```

**Comment by guyromb** [comment]
Is there anything that might be useful to check why is this happening? Anyone able to re-produce?
Thanks

**Comment by SantoDE** [comment]
Hey @guyromb,

I took some time to try and reproduce your issue today, however, I can't.

I deployed the `containous/whoamitcp` service like that:

```
---
kind: Deployment
apiVersion: apps/v1
metadata:
  name: whoami-tcp
  namespace: whoami
  labels:
    app: whoami-tcp
spec:
  replicas: 2
  selector:
    matchLabels:
      app: whoami-tcp
  template:
    metadata:
      labels:
        app: whoami-tcp
    spec:
      containers:
        - name: whoami-tcp
          image: containous/whoamitcp
          ports:
            - name: tcp
              containerPort: 8080
              protocol: TCP
---
apiVersion: v1
kind: Service
metadata:
  name: whoami-tcp
  namespace: whoami
  labels:
    app: whoami-tcp
spec:
  ports:
    - protocol: TCP
      name: whoami
      port: 8080
  selector:
    app: whoami-tcp
```

Afterwards, I created the following IngressRouteTCP Object

```
---
apiVersion: traefik.containo.us/v1alpha1
kind: IngressRouteTCP
metadata:
  name: whoami-tcp
  namespace: whoami
spec:
  entryPoints:
    - tcptest
  routes:
    - match: HostSNI(`*`)
      kind: Rule
      services:
      - name: whoami-tcp
        port: 8080
```

Connecting to whoami through Traefik on the dedicated Entrypoint for that service, showed that loadbalancing is working:

First req:
```
nc -n 127.0.0.1 4711
WHO
Hostname: whoami-tcp-7b489d4fc8-zkfxk
IP: 127.0.0.1
IP: 10.42.1.11
```

Seconc req:
```
nc -n 127.0.0.1 4711
WHO
Hostname: whoami-tcp-7b489d4fc8-nntzr
IP: 127.0.0.1
IP: 10.42.0.6
```

As you can see, the Hostname changed as it was answered by the different pods.

Therefore, I'll close that issue.

**Comment by traefiker** [comment]
Closed by #6979.
